frontend/app/routes/public/apply/$id/adult-child/verify-email.tsx

```diff
@@ -0,0 +1,313 @@
+      verificationCode: z
+        .string()
+        .trim()
+        .min(1, t('apply-adult-child:verify-email.error-message.verification-code-required'))
+        .transform(extractDigits)
+        .superRefine((val, ctx) => {
+          if (state.verifyEmail && state.verifyEmail.verificationAttempts >= MAX_ATTEMPTS) {
+            ctx.addIssue({ code: z.ZodIssueCode.custom, message: t('apply-adult-child:verify-email.error-message.verification-code-max-attempts'), path: ['verificationCode'] });
+          }
+        }),
+    });
+
+    const parsedDataResult = verificationCodeSchema.safeParse({
+      verificationCode: formData.get('verificationCode') ?? '',
+    });
+
+    if (!parsedDataResult.success) {
+      return data({ errors: transformFlattenedError(parsedDataResult.error.flatten()) }, { status: 400 });
+    }
+
+    // Check if the verification code matches
+    if (state.verifyEmail && state.verifyEmail.verificationCode !== parsedDataResult.data.verificationCode) {
+      const verificationAttempts = state.verifyEmail.verificationAttempts + 1;
+
+      saveApplyState({
+        params,
+        session,
+        state: {
+          verifyEmail: {
+            ...state.verifyEmail,
+            verificationAttempts,
+          },
+        },
+      });
+
+      return { status: 'verification-code-mismatch' } as const;
+    }
+
+    if (state.verifyEmail) {
+      saveApplyState({
+        params,
+        session,
+        state: {
+          verifyEmail: {
+            ...state.verifyEmail,
+            verificationAttempts: 0,
+          },
+          emailVerified: true,
+        },
+      });
+    }
+
+    if (state.editMode) {
+      return redirect(getPathById('public/apply/$id/adult-child/review-adult-information', params));
+    }
+
+    return redirect(getPathById('public/apply/$id/adult-child/dental-insurance', params));
+  }
+}
+
+export default function ApplyFlowVerifyEmail({ loaderData, params }: Route.ComponentProps) {
+  const { t } = useTranslation(handle.i18nNamespaces);
+  const { email, editMode } = loaderData;
+  const [showDialog, setShowDialog] = useState(false);
+
+  const fetcher = useFetcher<typeof action>();
+  const isSubmitting = fetcher.state !== 'idle';
+
+  const fetcherStatus = typeof fetcher.data === 'object' && 'status' in fetcher.data ? fetcher.data.status : undefined;
+  const errors = typeof fetcher.data === 'object' && 'errors' in fetcher.data ? fetcher.data.errors : undefined;
+  const errorSummary = useErrorSummary(errors, { verificationCode: 'verification-code' });
+
+  const communicationLink = <InlineLink routeId="public/apply/$id/adult-child/communication-preference" params={params} />;
+
+  useEffect(() => {
+    if (fetcherStatus === 'verification-code-sent') {
+      setShowDialog(true);
+    }
+  }, [fetcherStatus, fetcher.data]);
+
+  return (
+    <>
+      <div className="my-6 sm:my-8">
+        <Progress value={76} size="lg" label={t('apply:progress.label')} />
```

Comment:
Based on the Progress bar percentages document

```suggestion
        <Progress value={65} size="lg" label={t('apply:progress.label')} />
```